### PR TITLE
[KeyVault] Command fixes

### LINKED ISSUE #2332: Keyvault: inconsistent help information
In the shorter version of the help, which shows after missing parameter, the vault name parameter is described as `VAULE_BASE_URL`. However in the full help text, it is decribed as the `name of the vault`.
```
(env) ✔ ~/code/azure/azure-cli [master|✔] 
13:33 $ az keyvault secret list
az keyvault secret list: error: the following arguments are required: --vault-name
usage: az keyvault secret list [-h] [--output {json,tsv,table,jsonc}]
                               [--verbose] [--debug] [--query JMESPATH]
                               --vault-name VAULT_BASE_URL
                               [--maxresults MAXRESULTS]
(env) ✘-2 ~/code/azure/azure-cli [master|✔] 
13:33 $ az keyvault secret list -h

Command
    az keyvault secret list: List secrets in the specified vault.

Arguments
    --vault-name [Required]: Name of the key vault.
    --maxresults           : Maximum number of results to return in a page. If not specified the
                             service will return up to 25 results.

Global Arguments
    --debug                : Increase logging verbosity to show all debug logs.
    --help -h              : Show this help message and exit.
    --output -o            : Output format.  Allowed values: json, jsonc, table, tsv.  Default:
                             table.
    --query                : JMESPath query string. See http://jmespath.org/ for more information
                             and examples.
    --verbose              : Increase logging verbosity. Use --debug for full debug logs.
```

### LINKED ISSUE #2361: az keyvault create command parameters aren't documented/don't work as expected
The following parameters suggest they are flags but they are not.  The help does not describe their meaning or intended values

`                          [--enabled-for-disk-encryption ENABLED_FOR_DISK_ENCRYPTION]
                          [--enabled-for-template-deployment ENABLED_FOR_TEMPLATE_DEPLOYMENT]
                          [--enabled-for-deployment ENABLED_FOR_DEPLOYMENT]`

`    --enabled-for-deployment         : Property to specify whether Azure Virtual Machines are
                                       permitted to retrieve certificates stored as secrets from the
                                       key vault.
    --enabled-for-disk-encryption    : Property to specify whether Azure Disk Encryption is
                                       permitted to retrieve secrets from the vault and unwrap keys.
    --enabled-for-template-deployment: Property to specify whether Azure Resource Manager is
                                       permitted to retrieve secrets from the key vault.`

### LINKED ISSUE #2362: KeyVault create resource group required but not marked as such
Resource group is required but not marked as so in keyvault create.

```
$ az keyvault create --help

Command
    az keyvault create: Create a key vault.
        Default permissions are created for the current user unless the --no-self-perms flag is
        specified.

Arguments
    --location -l          [Required]: Location.
    --name -n              [Required]: Name of the key vault.
    --enabled-for-deployment         : Property to specify whether Azure Virtual Machines are
                                       permitted to retrieve certificates stored as secrets from the
                                       key vault.
    --enabled-for-disk-encryption    : Property to specify whether Azure Disk Encryption is
                                       permitted to retrieve secrets from the vault and unwrap keys.
    --enabled-for-template-deployment: Property to specify whether Azure Resource Manager is
                                       permitted to retrieve secrets from the key vault.
    --no-self-perms                  : If specified, don't add permissions for the current user in
                                       the new vault.
    --resource-group -g              : Name of resource group.
    --sku                            : SKU details.  Allowed values: premium, standard.  Default:
                                       standard.
    --tags                           : Space separated tags in 'key[=value]' format. Use '' to clear
                                       existing tags.

Global Arguments
    --debug                          : Increase logging verbosity to show all debug logs.
    --help -h                        : Show this help message and exit.
    --output -o                      : Output format.  Allowed values: json, jsonc, table, tsv.
                                       Default: json.
    --query                          : JMESPath query string. See http://jmespath.org/ for more
                                       information and examples.
    --verbose                        : Increase logging verbosity. Use --debug for full debug logs.
$ az keyvault create --name debekoe-kv1 -l westus
Parameter 'resource_group_name' can not be None.

```

### LINKED ISSUE #2418: Update issuer record --enabled flag doesn't allow setting enabled to false
Update issuer record
(--enabled flag doesn't allow setting enabled to false, does allow setting enabled to true when issuer is initially created disabled)
for example: 
az keyvault certificate issuer update --issuer-name MyDigiCert --vault-name cli20vault --enabled true --password <api-key> --account-id 117808

### LINKED ISSUE #2420: Key Vault: Create command accepts any value for --enabled-for-* flag as true
az keyvault create --resource-group demo9300 --name demo9371 --sku premium --location  westus **--enabled-for-deployment 0 --enabled-for-disk-encryption 0** --enabled-for-template-deployment 1 --tags tag1=vaule1 tag2=value2

Irrespective of the value provided for parameters --enabled-for-* it's treated as 'true'.

### LINKED ISSUE #2421: Key Vault: Cannot set Enabled-For-* options AFTER a key vault is created.
az keyvault set-policy 

This command does not accept --enabled-for-* options. Equivalent PSH command allows user to set the EnabledFor* options.

For example,
Set-AzureRmKeyVaultAccessPolicy [-VaultName] <String> [[-ResourceGroupName] <String>] [-EnabledForDeployment] [-EnabledForDiskEncryption]
[-EnabledForTemplateDeployment] [-PassThru] [-Confirm] [-WhatIf] [<CommonParameters>]

### PR DESCRIPTION
**Pull request by tjprescott**
- Fixes #2332. Adds METAVAR for vault
- Fixes #2362. Makes resource group required for keyvault create. 
- Adds reusable method in core to get location from resource group. Updates references. Makes location optional for keyvault create.
- Fixes #2420. Fixes #2361. Adds "three_state_flag" to core parameters to allow true/false or flag-like behavior (avoids breaking changes and allows create/update consistency).
- Closes #2421. Adds convenience parameters for this update command.
- Fixes #2418. Corrects bug with enabled/disabled flags. Makes keyvault certificate create idempotent.

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/commands/parameters.py [hunk 1]
```diff
@@ -107,12 +107,33 @@ def enum_choice_list(data):
...
         return next((x for x in choices if x.lower() == value.lower()), value) if value else value
     params = {
-        'choices': CaseInsenstiveList(choices),
+        'choices': CaseInsensitiveList(choices),
         'type': _type
     }
     return params
 
 
+def three_state_flag(none_default=True, use_enabled_disabled=False):
```
Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/commands/parameters.py]: Don't think `none_default` and `use_enabled_disabled` are clear names.
Suggest a more descriptive name or method comments describing the args.

File: src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py [hunk 2]
```diff
@@ -216,13 +215,14 @@ def get_default_policy(client, scaffold=False): #pylint: disable=unused-argument
...
-def create_keyvault(client, resource_group_name, vault_name, location, #pylint:disable=too-many-arguments
+def create_keyvault(client, resource_group_name, vault_name, location=None, #pylint:disable=too-many-arguments
                     sku=SkuName.standard.value,
                     enabled_for_deployment=None,
                     enabled_for_disk_encryption=None,
                     enabled_for_template_deployment=None,
-                    no_self_perms=False,
+                    no_self_perms=None,
                     tags=None):
+    from azure.mgmt.keyvault.models import VaultCreateOrUpdateParameters
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py]: Is this import used?

Comment (derekbekoe) [inline src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py]: Same in other places I think

Comment (tjprescott) [inline src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py]: If it weren't, it wouldn't pass the CI :)

**Comment by tjprescott** [comment]
@derekbekoe I simplified the "three_state_flag" per your feedback. Please take a look.

**Pull request merged**